indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -8,951 +8,200 @@
-    return (
-      !attributes ||
-      (!room && !this.newRoom) ||
-      !roomAttributes ||
-      !roomEquipment ||
-      !roomAvailability
-    );
-  }
-
-  async fetchAttributes() {
-    let response;
-    try {
-      response = await indicoAxios.get(fetchAttributesURL());
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    this.setState({attributes: response.data});
-  }
-
-  async fetchRoomData(roomId) {
-    const reqs = [
-      this.fetchDetailedRoom(roomId),
-      this.fetchRoomAttributes(roomId),
-      this.fetchRoomAvailability(roomId),
-      this.fetchRoomEquipment(roomId),
-    ];
-    this.setState(Object.assign(...(await Promise.all(reqs))));
-  }
-
-  async fetchDetailedRoom(roomId) {
-    let response;
-    try {
-      response = await indicoAxios.get(roomURL({room_id: roomId}));
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    return {room: camelizeKeys(response.data)};
-  }
-
-  async fetchRoomAttributes(roomId) {
-    let response;
-    try {
-      response = await indicoAxios.get(fetchRoomAttributesURL({room_id: roomId}));
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    return {roomAttributes: response.data};
-  }
-
-  async fetchRoomAvailability(roomId) {
-    let response;
-    try {
-      response = await indicoAxios.get(fetchRoomAvailabilityURL({room_id: roomId}));
-    } catch (error) {
-      handleAxiosError(error);
-      return;
-    }
-    const roomAvailability = camelizeKeys(response.data);
-    roomAvailability.nonbookablePeriods.forEach(period => {
-      period.key = shortid.generate();
-    });
-    roomAvailability.bookableHours.forEach(hours => {
-      hours.key = shortid.generate();
-    });
-
-    return {roomAvailability};
-  }
+  const isNewRoom = useCallback(() => roomId === undefined, [roomId]);
 
-  async fetchRoomEquipment(roomId) {
+  const fetch = useCallback(async (url, params = {}) => {
     let response;
     try {
-      response = await indicoAxios.get(fetchRoomEquipmentURL({room_id: roomId}));
+      response = await indicoAxios.get(url(params));
     } catch (error) {
       handleAxiosError(error);
       return;
     }
-    return {roomEquipment: camelizeKeys(response.data)};
-  }
-
-  handleCloseModal = async () => {
-    const {
-      afterCreation,
-      onClose,
-      actions: {fetchEquipmentTypes, fetchRoom, fetchRoomDetails, fetchRoomPermissions},
-    } = this.props;
-    const {wasEverSaved} = this.state;
-    // eslint-disable-next-line react/destructuring-assignment
-    const roomId = this.newRoom ? this.state.newRoomId : this.props.roomId;
-    this.setState({closing: true});
-    if (roomId !== null && wasEverSaved) {
-      await Promise.all([
-        fetchEquipmentTypes(),
-        fetchRoom(roomId),
-        fetchRoomPermissions(roomId),
-        fetchRoomDetails(roomId, true),
-      ]);
-    }
```

Comment:
It looks like you didn't add this back in the new code. It's required to update the data in redux when editing a room from outside the admin area.

Reply:
I was wondering what that was for. Eventually, I noticed there are different room fetch calls for admin and non-admin data. Thanks!

Reply:
Exactly - the admin data is freshly loaded all the time (since you REALLY don't want to accidentally edit a room using stale data), while the data for users is loaded once on page load, and then never reloaded (except when you are an admin and enter+leave the admin area, or edit a room from the normal area)